pilot/pkg/model/gateway.go

```diff
@@ -323,6 +319,9 @@ func resolvePorts(number uint32, instances []*ServiceInstance, legacyGatewaySele
 			ports[w.Endpoint.EndpointPort] = struct{}{}
 		}
 	}
+	if len(ports) > 1 {
+		log.Warnf("gateway: found %d services on port %d with different target port: %v", len(ports), number, ports)
```

Comment:
why warn?

Reply:
it's totally valid case to have multiple

Reply:
Copied from https://github.com/istio/istio/pull/34978/files#diff-7abd8e9bc7226623690e85b5acc0280def2d72df7960b1324349b08dcb186ca1L100

Do you mean it is valid below:
2 services for gateway.

1. 80-->8080

2. 80-->8081

Reply:
In the old code it was invalid. In the new code it is valid

Reply:
Then in this case, which port will be used as listener/route

Reply:
removed